### ISSUE TITLE
warning CS8618 reported for uninitialized field marked [MaybeNull]

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#37511]
```C#
#nullable enable
#pragma warning disable 169
#pragma warning disable 414

using System.Diagnostics.CodeAnalysis;

class C<T>
{
    [MaybeNull] T F1;
    [MaybeNull] object F2;
}
```
```
(9,19): warning CS8618: Non-nullable field 'F1' is uninitialized.
(10,24): warning CS8618: Non-nullable field 'F2' is uninitialized.
```

**Comment by RikkiGibson** [comment]
@cston do we expect this to not warn for fields, properties and field-like events?

**Comment by cston** [comment]
@RikkiGibson, the compiler should not warn for fields or properties where the type is marked `[MaybeNull]`. It doesn't look like the attribute can be applied to event declarations.

**Comment by RikkiGibson** [comment]
You're right, it's not allowed to put it on event declarations. I can't think of a scenario where it would be useful to do so.

**Comment by sharwell** [comment]
👍 this is frustrating for the case of unconstrained generic types, for which there is currently no working alternative.

**Comment by jcouv** [comment]
This is currently by design. Nullable attributes only affect callers at the moment.

**Comment by RikkiGibson** [comment]
We have been bit-by-bit changing that, though. See `NullableWalker.ApplyLValueAnnotations` and its call sites.
